File: python/pig_latin.py
import re

def translate(word_or_phrase):
  phrase = word_or_phrase.split(" ")
  result = []
  vowel = '^[aeiou]'
  consonants = '^[^aeiou]+'
  consonant_qu = '^[^aeiou]qu'
  q = '^qu'
  # print(re.search('^[aeiou]',word_or_phrase))
  for i in phrase:
    con = re.search(consonants,i)
    qu = re.search(q,i)
    con_qu = re.search(consonant_qu,i)
    if re.search(vowel,i):
      result.append(i + "ay")
    elif con_qu:
      result.append(i[con_qu.end():]+i[con_qu.start():con_qu.end()]+"ay")
    elif qu:
      result.append(i[qu.end():]+ i[qu.start():qu.end()] + "ay")
    elif con:
      result.append(i[con.end():] + i[con.start():con.end()]+ "ay")
  return " ".join(result)

# print(translate('quapple apple'))

File: js/pigLatin.js
exports.translate = function(word) {
    const phrase = word.split(" ")
    let result = []
    const vowel = '^[aeiouAEIOU]'
    const consonants = '/^[^aeiou]*/g'
    const consonant_qu = '/^[^aeiou]*qu/g'
    const q = '^qu'

    for (const word of phrase) {
        let vow = word.search(vowel)
        let con = word.search(consonants) +1
        let qu = word.search(q)
        let conQu = word.search(consonant_qu)
        console.log(qu)

        if (vow !== -1){
            result.push(word+"ay")
        }
        else if (conQu !== -1){
            const slice = word.slice(conQu) + word.slice(0, conQu) + "ay";
            result.push(slice)
        }
        else if (qu !== -1){
            const slice = word.slice(qu+2) + word.slice(0, qu+2) + "ay";
            result.push(slice)
        }
        else if (con !== -1){
            const slice = word.slice(con) + word.slice(0, con) + "ay";
            result.push(slice)
        }
    }
    // console.log(result.join(" "))
    return result.join(" ")
};
